doc/dev/contributing.rst

```diff
@@ -0,0 +1,224 @@
+main autodE remote repository (``duartegroup/autode``). In most cases,
+you should contribute to autodE through a pull request from a fork.
+
+
+Several, smaller pull requests instead of one big PR
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Smaller pull requests (PRs) are reviewed faster, and more accurately. We
+therefore encourage contributors to keep the set of changes within a
+single pull request as small as possible. If your pull request modifies
+more than 5 files, and/or several hundred lines of code, you should
+probably break it down into two or more pull requests.
+
+
+Pull requests are more than code
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+A pull request is difficult to review without a description of context
+and motivation for the attached set of changes. Whenever you open a new
+pull request, please include the following information:
+
+-  **A title** that explicits the main change addressed by the pull
+   request. If you struggle to come out with a short and descriptive
+   title, this is an indication that your could (should?) be broken down
+   into smaller PRs.
+-  **A description** of the context and motivation for the attached set
+   of changes. *What is the current state of things?*, *Why should it be
+   changed?*.
+-  **A summary** of changes outlining the main points addressed by your
+   pull request, and how they relate to each other. Be sure to mention
+   any assumption(s) and/or choices that your made and alternative
+   design/implementaions that you considered. *What did you change or
+   add?* *How?*. *Anything you could have done differently? Why not?*.
+-  **Some advice for reviewers**. Explicit the parts of your changes on
+   which you would expect reviewers to focus their attention. These are
+   often parts that you are unsure about or code that may be difficult to
+   read.
+
+
+Draft pull requests
+~~~~~~~~~~~~~~~~~~~
+
+Draft pull requests are a way to signal to other developers that you are
+currently working on something and open for discussion about it. It’s
+also providing the development team a glimpse of future code reviews.
+
+Look out for the “Convert to draft” button on the right hand side pane
+when creating a pull request.
+
+
+Style guidelines
+----------------
+
+Enforcing code style in contributions is key to maintain a consistent
+code base.
+
+
+Formatting
+~~~~~~~~~~
+
+autodE’s code loosely follows `the PEP8
+guidelines <https://www.python.org/dev/peps/pep-0008/>`__ for code
+formatting. At the very least, we expect all contributors to use
+formatters like `Black <https://github.com/psf/black>`__,
+`autopep8 <https://github.com/hhatto/autopep8>`__ or
+`YAPF <https://github.com/google/yapf>`__.
+
+
+Naming
+~~~~~~
+
+1. Variables
+
+   -  Variable names should be ``snake_case``.
+
+2. Functions
+
+   -  Like variables, function names should be ``snake_case``.
+
+   -  Functions should always exit with an explicit ``return``
+      statement, even if means ``return None``.
```

Comment:
this makes me unhappy

Reply:
Functions should return something - change my mind 😉

Reply:
putting `return None` is functionally identical to not having it there

Reply:
but readability

Reply:
hmph